cpp/src/cider/include/cider/processor/ExpressionEvaluator.h

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright(c) 2022-2023 Intel Corporation.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#ifndef CIDER_EXPRESSION_EVALUATOR_H
+#define CIDER_EXPRESSION_EVALUATOR_H
+
+#include <memory>
+#include "cider/processor/BatchProcessor.h"
+#include "exec/nextgen/context/Batch.h"
+#include "substrait/algebra.pb.h"
+#include "substrait/extended_expression.pb.h"
+
+namespace cider::exec::processor {
+using ExpressionPtr = std::shared_ptr<::substrait::Expression>;
+using ExpressionVector = std::vector<ExpressionPtr>;
+
+using Batch = cider::exec::nextgen::context::Batch;
```

Comment:
use `Batch` rather than `ArrowArray/ArrowSchema` as interface?

Reply:
Use batch would be more convenient than to use pair of ArrowArray/ArrowSchema. we should unify to use Batch instead of array and schema.

Reply:
users don't want to learn a new type, in the case of having `Arrow` already.

Reply:
I saw we use the new invent  type ' cider::exec::nextgen::context::Batch',  may be  use the [ArrowBatch ](https://arrow.apache.org/docs/cpp/api/table.html#_CPPv4N5arrow11RecordBatchE)   is make sense.

Reply:
`RecordBatch` is C++ API. we should not use that.

Reply:
Any reason we don't want to use the C++ API instead of C ABI?

Reply:
updated